### ISSUE TITLE
default installation forced the hijacked of root webserver, lack of information

### ISSUE THREAD

**Issue by mckaygerhard** [opensourcepos/opensourcepos#1716]
### Issue / Question / Bug

- [x] I'm reporting an issue of an unmodified OSPOS installation
- [ ] I checked open and closed issues database and no similar issue was already discussed (please make sure you searched!)
- [x] I read the README, WHATS_NEW and UPGRADE
- [x] I read the FAQ (https://github.com/jekkos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP has php-gd, php-intl, sockets and etc. installed)
- [x] I read the wiki
- [ ] I ran any database upgrade scripts (e.g. database/2.4_to_3.0.sql), and migrating function
- [ ] I'm aware the latest master could be a development version and therefore not stable
- [x] I know the version of OSPOS and git commit hash (check the footer of your OSPOS), the name and version of OS, Web server, PHP and MySQL and will add them to my issue report

### Installation information

- OSPOS version is: 3.1.1 stable
- OSPOS git commit hash is: 
- PHP version is:   5.6
- MySQL or MariaDB version is:   MariaDB 10.1
- OS and version is:   Linux Devuan
- WebServer is:   Apache 2.4 and/or Lighttpd

### Expected behaviour

as template said "i ran a unmodified version of.."  follow the install guide and then run from a subpath of web server root EVENT in the hole root web server

### Actual behaviour

1. must tune up two files, 
2. installation guide forced my webserver complety server only the ospos
3. there's not right info about neither in ther README or wiki
4. must search inside the application to solve
5. **default installation must not to be very intrusive**
6. default installation are not OS agnostic, assumed a service unique

### Steps to reproduce the issue

https://github.com/opensourcepos/opensourcepos/issues?q=is%3Aissue+install+directory+is%3Aclosed

**Comment by daN4cat** [comment]
Not sure I follow... I suppose you installed OSPOS in a subdir pointed by a subdomain if you run something in your root (e.g. your website).

**Comment by mckaygerhard** [comment]
in fact i installed in a subdir, but not in root webserver, due more secure and professional wway i setup a aliasing.. from /usr/share/web/ospos/public to "ospos" (so then http://<ip>/ospos)
the "pointed by a subdomain" are not related!

**Comment by nerv55** [comment]
same here

**Comment by daN4cat** [comment]
Current installation is already protected because public exposes only the minimum required. If you start playing with alias pointing stuff in different ways, then you are on your own.

**Comment by daN4cat** [comment]
And .htaccess files protect ospos as ospos/public. In ospos/public if not the root you need to change one line of .htaccess.

**Comment by daN4cat** [comment]
Also please read #920, it might give some hints.

**Comment by mckaygerhard** [comment]
@daN4cat you misc-understand i use an aliasing! so that "novice" .htaccess protection are unuseless, and you think my problem its the specific comment "920#issuecomment-254434793" ( #920#issuecomment-254434793 ) that are not the problem really

must reopen this issue or i must mae a new .. my instalation are outside of the root webserver so that "stupid" protection are not related

i talking about the ALL the installation are "hard-coded" to a unique path respect the ".htaccess file"

also very hard-code and realted to apache2

**Comment by daN4cat** [comment]
Good, so you have something to contribute.
The fact that you want to work from apache config with alias indeed does not require .htaccess.

Let us know the solution whdn everything works fine.

**Comment by mckaygerhard** [comment]
hi i investigate and seem some code paths are "forced" to the root of the webserver.. i remenber that in older php-pos in some views the "anchor" and "url_base" helper functions are not used.. and hardcoded paths are used instead..

so today i'll try a new installation but **how can i enable more loggin,**? when i set to 4 the level of log, the only entries are "initialiation, doing, etc etc" and apache does not show more..

**Comment by daN4cat** [comment]
Eh yes..  useless and stupid things...

**Comment by daN4cat** [comment]
But you are not a novice and pretty sure you can turn it around at a blink of an eye.

**Comment by mckaygerhard** [comment]
@daN4cat i know you mean that i must put "echoes" aqnd "alerts".. but i mean for some production cases? well in any case...

**Comment by mckaygerhard** [comment]
the rewrite rule of the public are full of "examples" but none o these examples are in sync with the revwrite rule in root ospos directory.. 

i'm not expert but if i change the url of the example by the ospos subdir path, never will work, due all the controllers has "redirect(controllername)" but controllername must be under base_path that not are same .. and to solve all of you made the root rewirte rule that send all the request to the public subpath of the webserver

very complicated and confusing.. the rewrite rule in root are used to not use the index.php for "nice" urls.. all are "more hardcoded" rather than "configured"

for able to use subpaths corretly i must let the index.php in sync with the rewrite rules.. of course defined the index file, 

but the original rewirte rules this only works with apache2, a webserver with low performance in large production sites, that actually uses lighttpd or nginx

projects like wordpress and magento can use easyle secure urls.. but opensourcepos have a very complicated and not more secure way to redirect and use a public directory.. 

in conclusion sounds that try to reinventing the well, and the instalation are very very complicated...confusin and ilogically, with a excuse called "security"

**Comment by daN4cat** [comment]
@mckaygerhard free to rewrite the code, nobody forces you to use this one. It is what it is and fine for most of us, CodeIgniter community, php composer based installation... I see many people happy with it.

**Comment by mckaygerhard** [comment]
umm i now see why putin has right! people that u said "are happy with composer" think the always will be "online"... well baby not all are onlyne, i must download the complete source and install offline at my home.. pufff that's the reason why in 3rt world countr are so many ilegal software

**Comment by daN4cat** [comment]
I would say that OSPOS is part of a conspiracy for sure, the big brother is controlling our brains.... it's all written on Fakebook.

**Comment by mckaygerhard** [comment]
the same people that are "happy with composer" its the same that spends many hours in "faisbuk" (Fakebook?), witout those, there's no real conspirations.. 

"oh great composer, all are in the network", i not have network men! pfff

**Comment by odiea** [comment]
If you use the bintray version you would not need composer .

**Comment by mckaygerhard** [comment]
@odiea what do you mean as bintray version? and i want to work directly with the git code, i try to generate a specific country instalation, i can see that the wiki pages are very outdated and inconsistent!

**Comment by odiea** [comment]
On the front page read the local install. These are ready to use versions. Stable 3.1.1 Unstable 3.2.0.

**Comment by mckaygerhard** [comment]
i read, got the unstable versions, where i can find info about the source respect the "bintray" version, why i canot deploy  the sources directly from the git code? i need more info!

**Comment by odiea** [comment]
That was changed about 6 or 8 months ago. If you are a developer with all the necessary tools then you can use the github version.

**Comment by mckaygerhard** [comment]
sorry for insist @odiea i try to do something like developer, but all my resources are offline ok! and also i need to code directly, must i use a deploy every change in the git code? why my direct code does not work!? i got a RaceCore error "class not found" when run login, but all the vendor files are in it

**Comment by odiea** [comment]
All I do is use notepad++ and change my local install until it works the way I want it to. When you are finished and want to submit a pull request then you can update your github branch with what you have done so far.

**Comment by mckaygerhard** [comment]
i not have network in "my local install" i'm at the office, and i not use "notepad++" i tarring the complete git repo go to office made pull or push and return to my home at night and so go on.. every day!

its a pain made that every day, so i cannot get my git source directly in hiawatta due i must use a "deploy" to see running named "bintray"

do you understand what i taking about!?
